views/setup-wizard/attach-to.php

```diff
@@ -10,28 +10,18 @@
 	// load invoice to reuse method to get wc emails
 	$invoice = wcpdf_get_invoice( null );
 	$wc_emails = $invoice->get_wc_emails();
-	foreach ( $wc_emails as $email_id => $name ) {
-		if ( ! empty( $current_settings['attach_to_email_ids'][$email_id] ) ) {
-			$checked = 'checked';
-		} else {
-			$checked = '';
-		}
-		printf(
-			'<tr>
-				<th>
-					<input type="hidden" value="" name="wcpdf_settings[wpo_wcpdf_documents_settings_invoice][attach_to_email_ids][%1$s]">
-					<input id="%1$s" type="checkbox" %3$s name="wcpdf_settings[wpo_wcpdf_documents_settings_invoice][attach_to_email_ids][%1$s]" value="1">
-				</th>
-				<td>
-					<label for="%1$s" class="checkbox">%2$s</label>
-				</td>
-			</tr>',
-			esc_attr( $email_id ),
-			esc_html( $name ),
-			esc_attr( $checked )
-		);
 
-	}
+	$attach_to_setting = array(
+		'option_name'     => 'wcpdf_settings[wpo_wcpdf_documents_settings_invoice]',
+		'id'              => 'attach_to_email_ids',
+		'options'         => $wc_emails,
+		'multiple'        => true,
+		'enhanced_select' => true,
+		'default'         => isset( $current_settings['attach_to_email_ids'] ) ? array_keys( array_filter( $current_settings['attach_to_email_ids'], function( $value ) { return $value === '1'; } ) ) : array(),
```

Comment:
I'm unsure if we should provide a default here, as it's not something we do in the settings.